README.md

```diff
@@ -2,92 +2,707 @@
+
+### Token Validation
+
+PelicanFS automatically validates tokens to ensure they:
+- Have not expired
+- Have the correct audience and issuer
+- Have the necessary scopes for the requested operation
+- Are authorized for the specific namespace path
+
+## Integration with Data Science Libraries
+
+PelicanFS integrates seamlessly with popular Python data science libraries.
+
+### Using with xarray and Zarr
+
+PelicanFS works great with xarray for reading Zarr datasets:
+
+```python
+import xarray as xr
+
+# Method 1: Using the scheme directly (recommended - simplest)
+ds = xr.open_dataset('osdf:///ospool/uc-shared/public/dataset.zarr', engine='zarr')
+
+# Method 2: Using PelicanMap (useful for multiple datasets or custom configurations)
+from pelicanfs.core import PelicanFileSystem, PelicanMap
+pelfs = PelicanFileSystem("pelican://osg-htc.org")
+zarr_store = PelicanMap('/ospool/uc-shared/public/dataset.zarr', pelfs=pelfs)
+ds = xr.open_dataset(zarr_store, engine='zarr')
+
+# Method 3: Opening multiple datasets with PelicanMap
+file1 = PelicanMap("/ospool/uc-shared/public/file1.zarr", pelfs=pelfs)
+file2 = PelicanMap("/ospool/uc-shared/public/file2.zarr", pelfs=pelfs)
+ds = xr.open_mfdataset([file1, file2], engine='zarr')
+```
+
+### Using with PyTorch
+
+PelicanFS can be used to load training data for PyTorch:
+
+```python
+import torch
+from torch.utils.data import Dataset
+import fsspec
+
+class PelicanDataset(Dataset):
+    def __init__(self, file_paths):
+        self.file_paths = file_paths
+
+    def __len__(self):
+        return len(self.file_paths)
+
+    def __getitem__(self, idx):
+        # Read file using fsspec
+        data = fsspec.cat(self.file_paths[idx])
+        # Process your data here
+        return data
+
+# Method 1: Using fsspec with schemes (recommended)
+files = fsspec.glob('osdf:///ospool/uc-shared/public/training/data/**/*.bin')
+dataset = PelicanDataset(files)
+dataloader = torch.utils.data.DataLoader(dataset, batch_size=32)
+
+# Method 2: Using PelicanFileSystem directly (for more control)
+from pelicanfs.core import PelicanFileSystem
+pelfs = PelicanFileSystem("pelican://osg-htc.org")
+files = pelfs.glob('/ospool/uc-shared/public/training/data/**/*.bin')
+# Add scheme prefix to paths for fsspec.cat
+files_with_scheme = [f'pelican://osg-htc.org{f}' for f in files]
+dataset = PelicanDataset(files_with_scheme)
+dataloader = torch.utils.data.DataLoader(dataset, batch_size=32)
+```
+
+### Using with Pandas
+
+Read CSV and other tabular data formats:
+
+```python
+import pandas as pd
+import fsspec
+
+# Method 1: Using fsspec.open with schemes (recommended)
+with fsspec.open('osdf:///ospool/uc-shared/public/data.csv', 'r') as f:
+    df = pd.read_csv(f)
+
+# Or read directly with pandas (if pandas supports the scheme)
+df = pd.read_csv('osdf:///ospool/uc-shared/public/data.csv')
```

Comment:
What is meant by "if pandas supports the scheme" -- supposedly it does if you have this example, right? How do I know whether it's supported?